### PR TITLE
Work towards multi target nuget

### PR DESCRIPTION
**Pull request by networkfusion**
packaged nuget is still a little strange.

## Types of changes

- [ ] Improvement (non-breaking change that improves a feature, code or algorithm)
- [ ] Bug fix (non-breaking change which fixes an issue with code or algorithm)
- [ ] New feature (non-breaking change which adds functionality to code)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Config and build (change in the configuration and build system, has no impact on code or features)
- [ ] Dependencies (update dependencies and changes associated, has no impact on code or features)
- [ ] Unit Tests (add new Unit Test(s) or improved existing one(s), has no impact on code or features)
- [ ] Documentation (changes or updates in the documentation, has no impact on code or features)

## Checklist:

- [ ] My code follows the code style of this project (only if there are changes in source code).
- [ ] My changes require an update to the documentation (there are changes that require the docs website to be updated).
- [ ] I have updated the documentation accordingly (the changes require an update on the docs in this repo).
- [ ] I have read the [CONTRIBUTING](https://github.com/nanoframework/.github/blob/master/CONTRIBUTING.md) document.
- [ ] I have tested everything locally and all new and existing tests passed (only if there are changes in source code).
- [ ] I have added new tests to cover my changes.

### CODE REVIEW

File: nanoFirmwareFlasher.Tool/nanoFirmwareFlasher.Tool.csproj [hunk 1]
```diff
@@ -80,6 +102,7 @@
 	<PropertyGroup>
 		<!-- set this to 'net6.0\' to package contents in this TFM folder  -->
 		<PackageTfmSubFolder>net6.0\</PackageTfmSubFolder>
+		<PackageTfmSubFolder>net472\</PackageTfmSubFolder>
```
[the remark this answers is not in the document]
Comment (josesimoes) [reply nanoFirmwareFlasher.Tool/nanoFirmwareFlasher.Tool.csproj:105]: Not sure where this is coming from... this doesn't look like the current version... 🤔 
nanoFirmwareFlasher.Tool.csproj has only 88 lines...

**Comment by josesimoes** [comment]
@networkfusion don't bother using Costura. That was the original approach and it stopped working as we added support for Linux and Mac. Without going into too much detail: the ST tool has a ... let's call it curious... requirement on the files location from each other (among other nits).

The current setup took dozens of hours (from myself and Laurent) of trial and testing on Win, Mac and Linux to get to the sweat spot for the packaging organization and having the tools work properly on all platforms. 
I won't claim it can not be improved. I really don't think there is any need to mess up with this at this point. There is no performance gain or better packaging to get out of this. 
Gentle reminder: if it's not broken don't fix it. 😉

**Comment by josesimoes** [comment]
And curious why the current NuGet is not "proper"... 🤔

**Comment by Ellerbach** [comment]
+1 with Jose on this. It's been a very very very very long journey to pack everything, have all included in the size limit as well, figuring out the directory structure that will work (yes, some may look illogical but somehow needed to have all platforms in one nuget). If you want to make any change on this, please test it on the following platforms:

* Windows (10 and 11)
* Linux x64 (Ubuntu or Debian)
* Mac with arm64 (M1), one of the latest OS is fine
* Mac with x64 (intel base), one of the latest OS is fine

And yes, it's been tested on all those platforms.
I'll not approve anything that has not been tested in those 4 configurations.

**Comment by networkfusion** [comment]
> And curious why the current NuGet is not "proper"... 🤔

Sorry, bad naming on my part.

**Comment by networkfusion** [comment]
> +1 with Jose on this. It's been a very very very very long journey to pack everything, have all included in the size limit as well, figuring out the directory structure that will work (yes, some may look illogical but somehow needed to have all platforms in one nuget). If you want to make any change on this, please test it on the following platforms:
> 
>     * Windows (10 and 11)
> 
>     * Linux x64 (Ubuntu or Debian)
> 
>     * Mac with arm64 (M1), one of the latest OS is fine
> 
>     * Mac with x64 (intel base), one of the latest OS is fine
> 
> 
> And yes, it's been tested on all those platforms. I'll not approve anything that has not been tested in those 4 configurations.

Okay, I will keep this open for the moment, because I am still trying to work out what has changed with the latest VS Code extension with people reporting it no longer works and this lib being the only one that has changed since it stopped working. But dont expect it to be merged.

**Pull request closed without merging**